test/hotspot/gtest/utilities/test_growableArray.cpp

```diff
@@ -663,3 +663,17 @@ TEST(GrowableArrayCHeap, find_from_end_if) {
     ASSERT_EQ(index, -1);
   }
 }
+
+TEST(GrowableArrayCHeap, ReturningReferencesWorksAsExpected) {
+  GrowableArrayCHeap<int, mtTest> arr(8, 8, -1); // Pre-fill with 8 -1s
+  int& x = arr.at_grow(9, -1);
+  x = 2;
+  EXPECT_EQ(2, arr.at(9));
+  x = arr.top();
```

Comment:
This is not using reference, right? I thought you can only use reference assignment when you declare the `x` variable. This here is using value, I think.

Reply:
Maybe that was your intention. But then you should have another test where you use `top` with reference.

Reply:
It wasn't my intention, I just fumbled the semantics when writing the tests, thanks for the catch :-).